main/src/main/scala/org/clulab/processors/Document.scala

```diff
@@ -58,6 +68,31 @@ class Document(val sentences: Array[Sentence]) extends Serializable {
     val h1 = mix(h0, orderedHash(sentences.map(_.ambivalenceHash)))
     finalizeHash(h1, 1)
   }
+
+  /** Adds an attachment to the document's attachment map */
+  def addAttachment(name: String, attachment: DocumentAttachment): Unit = {
+    if (attachments.isEmpty)
+      attachments = Some(new mutable.HashMap[String, DocumentAttachment]())
+    attachments.get += name -> attachment
+  }
+
+  /** Retrieves the attachment with the given name */
+  def getAttachment(name: String): Option[DocumentAttachment] = {
+    if (attachments.isEmpty) None
+    else attachments.get.get(name)
```

Comment:
I think these two lines can be rewritten as `attachments.flatMap(_.get(name))`.

Reply:
I will fix Mihai's code ;-)  Thanks for the careful reading.